hazelcast/src/test/java/com/hazelcast/quorum/QuorumTest.java

```diff
@@ -227,6 +229,49 @@ public boolean apply(Collection<Member> members) {
         }
     }
 
+    @Test
+    public void givenQuorumFunctionConfigured_whenImplementsHazelcastInstanceAware_thenHazelcastInjectsItsInstance() throws Exception {
+        TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(1);
```

Comment:
A HZ instance factory in a local variable looks suspicious to me. Are those instances shutdown properly?